pySOT/utils.py

```diff
@@ -390,3 +390,182 @@ def optimize(self):
                 population = new_population
 
         return best_individual, best_value
+
+POSITIVE_INFINITY = float("inf")
+
+def domination(fA, fB, M):
+    """Returns True if a fA dominates fB
+
+    :param fA: Vector A of objective functions
+    :param fB: Vector B of objective functions
+    :param M: Number of objective functions in A and B
+    :return: Boolean
+    """
+
```

Comment:
This is a bit ugly. What about:
`return np.all(np.less(fA, fB))`
if the inputs are numpy. Otherwise, do the same thing with the corresponding list operations.

Reply:
I kept the original code, because the code should stop as soon as the domination check is violated (break function). This may make the code more efficient for problems with many objectives.

Reply:
I prefer the more compact and Pythonic version, which will likely be faster as well due to interpreter overhead.

Reply:
Done.